### PR TITLE
Kana Vocabulary

### PR DESCRIPTION
**Pull request by UnicornsOnLSD**
## Description

This PR adds support for the new kana vocabulary type. It's mostly just copy pasted, as it follows the same structure as vocabulary but without reading-related elements.

- [x] New feature

## How has this been Tested?

I haven't thoroughly tested this, but everything decodes fine when requesting `subjects()`. I should probably look into adding some unit tests 🙃

### CODE REVIEW

File: Sources/WaniKani/Models/Subject.swift [hunk 1]
```diff
@@ -32,6 +32,7 @@ public enum Subject: ModelProtocol {
     case radical(Radical)
     case kanji(Kanji)
     case vocabulary(Vocabulary)
+    case kana_vocabulary(KanaVocabulary)
```
Comment (aaronsky) [inline Sources/WaniKani/Models/Subject.swift]: can you please make this `kanaVocabulary` just to keep things consistent?

File: Sources/WaniKani/WaniKani.swift [hunk 2]
```diff
@@ -116,7 +116,7 @@ public struct WaniKani {
                 return nil
             }
 
-            var keepTrying = false
+            var keepTrying = true
```
Comment (aaronsky) [inline Sources/WaniKani/WaniKani.swift]: might want to rebase on main just so this isn't in the diff

**Review by aaronsky — CHANGES_REQUESTED** [review]
Thank you for your contribution! If you wouldn't mind, please add a test for this to `Resources/SubjectsTests.swift`, then I'll activate CI for this PR.

File: Sources/WaniKani/Models/Subject.swift (hunk 1 above)
Comment (UnicornsOnLSD) [reply Sources/WaniKani/Models/Subject.swift]: Fixed in 786396d28fa2131ea0523c6f11465e8b52bf8eba

File: Sources/WaniKani/WaniKani.swift (hunk 2 above)
Comment (UnicornsOnLSD) [reply Sources/WaniKani/WaniKani.swift]: Rebased

**Comment by UnicornsOnLSD** [comment]
Added a test, follows the same format as the rest of the subject tests

**Review by aaronsky — APPROVED** [review]
thanks!

**Pull request merged**